http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/PersistentDiskComponent.scala

```diff
@@ -174,12 +180,11 @@ object persistentDiskQuery {
       .filter(_.name === name)
       .filter(_.destroyedDate === dummyDate)
 
-  private[db] def findByNameQuery(cloudContext: CloudContext, name: DiskName) =
+  private[db] def findActiveByIdWorkspaceQuery(workspaceId: WorkspaceId, id: DiskId) =
     tableQuery
-      .filter(
-        _.cloudContext === cloudContext.asCloudContextDb
-      )
-      .filter(_.name === name)
+      .filterOpt(Some(workspaceId))(_.workspaceId === _)
+      .filter(_.id === id)
```

Comment:
hmm... this query is a bit weird. If id itself is a primary key enforced by table unique constraints, and it's required (aka non-Optional) for this function, does filtering by these other things do anything? Seems a trivial optimization to just filter by ID.

Reply:
so I wrote it this way to enforce that the disk has to be from the specified workspace in the call, otherwise a user could put a workspace they have ownership over and the diskId of a disk outside that scope, however, I guess I could just check that the workspaceIds match after grabbing the disk in the getDisk function.

Reply:
Hmm... I'll leave it up to you. I would expect that if you ask sam 'does user A have permission to do B to resource C in workspace D' would trivially error if the resource is not in the workspace.